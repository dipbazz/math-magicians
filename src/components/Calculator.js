import React, { useState } from 'react';
import '../styles/Calculator.css';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

function Calculator() {
  const [total, setTotal] = useState('');
  const [next, setNext] = useState('');
  const [operation, setOperation] = useState(null);

  function handleClick(buttonName) {
    const calculation = calculate({
      total: total === 'Error' ? '' : total,
      next,
      operation,
    }, buttonName);

    setTotal(calculation.total);
    setNext(calculation.next);
    setOperation(calculation.operation);
  }

  function displayResult() {
    let result = '';

    if (total) {
      result += total;
    }
    if (operation) {
      result += operation;
    }
    if (next) {
      result += next;
    }
    return result;
  }

  return (
    <div className="flex">
      <h3>
        Let&#39;s do some math!
      </h3>
      <div className="Calculator">
        <Display result={displayResult()} />
        <ButtonPanel handleClick={(name) => handleClick(name)} />
      </div>
    </div>
  );
}

export default Calculator;
