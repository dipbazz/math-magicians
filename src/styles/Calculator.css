@import url('https://fonts.googleapis.com/css2?family=Farro:wght@300&display=swap');

* {
  font-family: 'Farro', sans-serif;
}

.Calculator {
  text-align: center;
  width: 360px;
  margin: auto;
  margin-top: 20px;
  background: #f1f2f6;
  padding: 20px 10px;
  border-radius: 10px;
}

.flex {
  display: flex;
}
