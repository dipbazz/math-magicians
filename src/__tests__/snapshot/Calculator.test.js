import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../../components/Calculator';

it('renders calculator correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
