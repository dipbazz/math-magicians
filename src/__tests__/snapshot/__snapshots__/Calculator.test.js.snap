// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders calculator correctly 1`] = `
<div
  className="flex"
>
  <h3>
    Let's do some math!
  </h3>
  <div
    className="Calculator"
  >
    <div
      className="display"
    >
      
    </div>
    <div>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="button  button--orange"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="button  button--orange"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="button  button--orange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button  button--orange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button  "
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="button button--span-2 button--orange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
